code/go/0chain.net/blobbercore/handler/storage_handler.go

```diff
@@ -480,6 +480,17 @@ func (fsh *StorageHandler) ListEntities(ctx context.Context, r *http.Request) (*
 		if clientID != allocationObj.OwnerID {
 			delete(result.Entities[idx], "path")
 		}
+
+		if child.Type == reference.DIRECTORY {
+			continue
+		}
+		// getting collaborators for sub dirs
+		collaborators, err := reference.GetCollaborators(ctx, child.ID)
+		if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
+			Logger.Error("Failed to get collaborators from refID", zap.Error(err), zap.Any("ref_id", dirref.ID))
+			return nil, err
+		}
+		result.Meta["collaborators"] = append(result.Meta["collaborators"].([]reference.Collaborator), collaborators...)
```

Comment:
should we only show them for owner like line 480?  `clientID != allocationObj.OwnerID`? try to check it with  @stewartie4

Reply:
updated